### PR TITLE
fix(openid redirect): redirect to user url after openid login

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-1340

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-server/1275)

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/security/UserRedirectBean.java [hunk 1]
```diff
@@ -205,4 +207,12 @@ public void captureCurrentView(@Observes NotLoggedInEvent event) {
         setUrl(urlUtil.getLocalUrl(request));
     }
 
+    public String redirect() {
+        try {
+            FacesNavigationUtil.redirect(FacesContext.getCurrentInstance(), getUrl());
+        } catch (IOException e) {
+            return "continue";
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/security/UserRedirectBean.java]: Why would an IOException happen here? Shouldn't we at least log something?

File: zanata-war/src/main/java/org/zanata/action/LoginAction.java [hunk 2]
```diff
@@ -141,12 +141,12 @@ public String login() {
...
     private String continueToPreviousUrl() {
         try {
-            FacesNavigationUtil.redirect(FacesContext.getCurrentInstance(),
-                    userRedirect.getUrl());
-        } catch (IOException e) {
+            urlUtil.redirectToInternal(userRedirect.getUrl());
+            return "continue";
+        } catch (Exception e) {
+            // already logged by urlUtil.redirectToInternal.
             return "dashboard";
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/action/LoginAction.java]: Sorry, I just realised what a terrible idea this is. Now we could be swallowing all sorts of exceptions, eg ClassNotFoundException, without logging anything.

**Comment by seanf** [comment]
:+1:

**Comment by djansen-redhat** [comment]
I seem to be getting this when logging in:
```
05:38:41,791 ERROR [org.zanata.exception.handler.AbstractExceptionHandler] (default task-43) exception happened in view: /openid.xhtml: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at javax.faces.application.NavigationCase.getToViewId(NavigationCase.java:311)
	at com.sun.faces.application.NavigationHandlerImpl.determineViewFromActionOutcome(NavigationHandlerImpl.java:1358)
	at com.sun.faces.application.NavigationHandlerImpl.findExactMatch(NavigationHandlerImpl.java:609)
	at com.sun.faces.application.NavigationHandlerImpl.getViewId(NavigationHandlerImpl.java:480)
	at com.sun.faces.application.NavigationHandlerImpl.handleNavigation(NavigationHandlerImpl.java:190)
	at com.sun.faces.application.NavigationHandlerImpl.handleNavigation(NavigationHandlerImpl.java:183)
	at org.apache.deltaspike.jsf.impl.config.view.navigation.ViewConfigAwareNavigationHandler.handleNavigation(ViewConfigAwareNavigationHandler.java:146)
	at org.apache.deltaspike.jsf.impl.scope.viewaccess.ViewAccessScopedAwareNavigationHandler.handleNavigation(ViewAccessScopedAwareNavigationHandler.java:51)
	at org.apache.deltaspike.jsf.impl.navigation.DeltaSpikeNavigationHandler.handleNavigation(DeltaSpikeNavigationHandler.java:77)
	at org.apache.deltaspike.jsf.impl.navigation.DeltaSpikeNavigationHandlerWrapper.handleNavigation(DeltaSpikeNavigationHandlerWrapper.java:47)
	at org.zanata.util.FacesNavigationUtil.handlePageNavigation(FacesNavigationUtil.java:52)
	at org.zanata.security.OpenIdPhaseListener.beforePhase(OpenIdPhaseListener.java:64)
	at com.sun.faces.lifecycle.Phase.handleBeforePhase(Phase.java:228)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:99)
	at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:219)
	at org.apache.deltaspike.jsf.impl.listener.request.DeltaSpikeLifecycleWrapper.render(DeltaSpikeLifecycleWrapper.java:111)
	at javax.faces.lifecycle.LifecycleWrapper.render(LifecycleWrapper.java:92)
	at org.apache.deltaspike.jsf.impl.listener.request.JsfClientWindowAwareLifecycleWrapper.render(JsfClientWindowAwareLifecycleWrapper.java:160)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:659)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.apache.deltaspike.servlet.impl.event.EventBridgeFilter.doFilter(EventBridgeFilter.java:59)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.apache.deltaspike.servlet.impl.produce.RequestResponseHolderFilter.doFilter(RequestResponseHolderFilter.java:63)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:226)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.zanata.servlet.DuplicateParamFilter.doFilter(DuplicateParamFilter.java:69)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.zanata.servlet.JavaMelodyFilter.doFilter(JavaMelodyFilter.java:65)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.zanata.servlet.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:76)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.zanata.servlet.RequestLoggingFilter.doFilter(RequestLoggingFilter.java:88)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:51)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:56)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:284)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:263)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:174)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:202)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:793)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

05:38:41,795 INFO  [org.zanata.ui.faces.FacesMessages] (default task-43) message to user (wid: -6142): An unexpected error has occurred. Please report this problem with details of what you were attempting.)
05:38:41,800 ERROR [org.jboss.as.txn] (default task-43) WFLYTX0003: APPLICATION ERROR: transaction still active in request with status 0
```

**Comment by djansen-redhat** [comment]
Also test:
- [ ] redirect to login from translate action (ZNTA-1348)

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested, looks good to me

**Pull request merged**